vms/platformvm/state/state.go

```diff
@@ -1785,20 +1786,80 @@ func (s *state) writeBlsKeyDiffs(height uint64, blsKeyDiffs map[ids.NodeID]*bls.
 	return nil
 }
 
+func (s *state) updateValidatorSet(
+	updateValidators bool,
+	valSetDiff map[subnetNodeKey]validatorStatusPair,
+	weightDiffs map[subnetNodeKey]*ValidatorWeightDiff,
+) error {
+	if !updateValidators {
+		return nil
+	}
+
+	for key, weightDiff := range weightDiffs {
+		var (
+			subnetID = key.subnetID
+			nodeID   = key.nodeID
+			val      = valSetDiff[key]
+			err      error
+		)
+
+		if weightDiff.Amount == 0 {
```

Comment:
We could replace these `if` statements with:

```go
		switch {
		case weightDiff.Amount == 0:
			// No weight change to record; go to next validator.
			continue
		case weightDiff.Decrease:
			err = s.cfg.Validators.RemoveWeight(subnetID, nodeID, weightDiff.Amount)
		case val.status == added:
			staker := val.validator
			err = s.cfg.Validators.AddStaker(
				subnetID,
				nodeID,
				staker.PublicKey,
				staker.TxID,
				weightDiff.Amount,
			)
		default:
			err = s.cfg.Validators.AddWeight(subnetID, nodeID, weightDiff.Amount)
		}
```

Reply:
Done